internal/terraform/node_resource_validate.go

```diff
@@ -313,40 +316,6 @@ func (n *NodeValidatableResource) validateResource(ctx EvalContext) tfdiags.Diag
 
 	diags = diags.Append(validateDependsOn(ctx, n.Config.DependsOn))
 
-	// Validate the provider_meta block for the provider this resource
-	// belongs to, if there is one.
-	//
-	// Note: this will return an error for every resource a provider
-	// uses in a module, if the provider_meta for that module is
-	// incorrect. The only way to solve this that we've found is to
-	// insert a new ProviderMeta graph node in the graph, and make all
-	// that provider's resources in the module depend on the node. That's
-	// an awful heavy hammer to swing for this feature, which should be
-	// used only in limited cases with heavy coordination with the
-	// Terraform team, so we're going to defer that solution for a future
-	// enhancement to this functionality.
-	/*
-		if n.ProviderMetas != nil {
-			if m, ok := n.ProviderMetas[n.ProviderAddr.ProviderConfig.Type]; ok && m != nil {
-				// if the provider doesn't support this feature, throw an error
-				if (*n.ProviderSchema).ProviderMeta == nil {
-					diags = diags.Append(&hcl.Diagnostic{
-						Severity: hcl.DiagError,
-						Summary:  fmt.Sprintf("Provider %s doesn't support provider_meta", cfg.ProviderConfigAddr()),
-						Detail:   fmt.Sprintf("The resource %s belongs to a provider that doesn't support provider_meta blocks", n.Addr),
-						Subject:  &m.ProviderRange,
-					})
-				} else {
-					_, _, metaDiags := ctx.EvaluateBlock(m.Config, (*n.ProviderSchema).ProviderMeta, nil, EvalDataForNoInstanceKey)
-					diags = diags.Append(metaDiags)
-				}
-			}
-		}
-	*/
-	// BUG(paddy): we're not validating provider_meta blocks on EvalValidate right now
```

Comment:
`ProviderMetas` are not being developed further, and we can removed these large block comments